File: 134/Version2014/Exercises/Ex34WMIDataReader.cs
            DataTableReader FileReader; DataTable FileTable;
            try
            {
                string strFileSystem, strFreeSpace, strSize, strMsg;
                FileTable = (DataTable)Dts.Variables["DiskInfo"].Value;
                FileReader = FileTable.CreateDataReader();
                while (FileReader.Read())
                {
                    strFileSystem = FileReader.GetString(0);
                    strSize = FileReader.GetString(1);
                    strFreeSpace = FileReader.GetString(2);
                    strMsg = "File system: " + strFileSystem + "\r\n";
                    strMsg += "Disk size: " + strSize + "\r\n";
                    strMsg += "Free space: " + strFreeSpace;
                    MessageBox.Show(strMsg);
                }
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
                Dts.TaskResult = (int)ScriptResults.Failure;
            }

File: 134/Version2016/Exercises/Ex34WMIDataReader.vb
        Dim FileReader As DataTableReader, FileTable As DataTable
        Try
            Dim strFileSystem As String, strFreeSpace As String, strSize As String, strMsg As String
            FileTable = CType(Dts.Variables("DiskInfo").Value, DataTable)
            FileReader = FileTable.CreateDataReader()
            While FileReader.Read
                strFileSystem = FileReader("FileSystem").ToString
                strSize = FileReader("Size").ToString
                strFreeSpace = FileReader("FreeSpace").ToString
                strMsg = "File system: " & strFileSystem & vbCrLf
                strMsg += "Disk size: " & strSize & vbCrLf
                strMsg += "Free space: " & strFreeSpace
                MsgBox(strMsg)
            End While
        Catch ex As Exception
            MsgBox(ex.Message)
            Dts.TaskResult = ScriptResults.Failure
            Return
        End Try
        FileReader.Close()